devops/providers/azure-devops/templates/infrastructure/README.md

```diff
@@ -0,0 +1,204 @@
+
+Once the resources have been provisioned within the Dev Int Environment, the test harness will run the automated unit and integration tests to validate the functionality end-to-end.
+
+The test harness will automatically tear down all provisioned resources to help minimize azure costs and duplicate cloud objects.
+
+### Build Trigger
+
+- Cobalt is a library of infrastructure manifests represented as Terraform templates(aka IAC: Infrastructure as Code).
+
+- Each template defines multiple composable building-block terraform modules and assembles them together to produce a larger system.
+
+- This pipeline's deployment process is built around a repo that holds the template to provision.
+
+- Operational changes are made to the running system by making commits on this repo.
+
+- Follow these [instructions](https://docs.microsoft.com/en-us/azure/devops/pipelines/repos/github?view=azure-devops&tabs=yaml#private-github-repository-or-azure-pipelines-in-a-private-project) to link the IAC repo to the build pipeline.
+
+- Here's some more [background](https://docs.microsoft.com/en-us/azure/devops/pipelines/repos/github?view=azure-devops&tabs=yaml#ci-triggers) on how CI triggers work in Azure DevOPS. 
+
+- The build pipeline is triggered on new commits made to any feature branch.
+
+#### Getting the source code
+
+The linked IAC git repo will be cloned and pulled onto the build agent machine. Here's some more [info](https://docs.microsoft.com/en-us/azure/devops/pipelines/repos/github?view=azure-devops&tabs=yaml#getting-the-source-code) on how that will operate from the perspective of the Azure unified pipeline.
+
+#### Git: Master Branch Policies
+
+We strongly recommend adding branch policies to help protect the master branch and configure mandatory validation builds to avoid simultaneous builds when merging into master.
+
+![image](https://user-images.githubusercontent.com/7635865/60196805-97c36680-9803-11e9-9fd0-7bedc34fc9ad.png)
+
+Here's some more [guidance](https://docs.microsoft.com/en-us/azure/devops/pipelines/repos/github?view=azure-devops&tabs=yaml#protecting-branches) on leveraging Azure DevOPS build validation checks with protected branch's.
+
+### Published Artifact
+
+The Terraform workspace directory is published as a pipeline build [artifact](https://docs.microsoft.com/en-us/azure/devops/pipelines/artifacts/build-artifacts?view=azure-devops&tabs=yaml#how-do-i-publish-artifacts) following the automated test step(s). This artifact ensures that the same manifest is applied across all release stages.
+
+## Release Stages
+
+### QA
+
+This environment is a lighter weight replica of staging and intended for manual quality assurance checkouts. Automated terraform tests are asserted against this environment, similar to DevInt.
```

Comment:
How can we verify that infrastructure changes don't break key application behaviors? Are these tests the infra tests for the template itself or are they application level integration tests?

Reply:
the tests are infra specific tests maintained by the technical operators that own the cloud subscriptions.